File: problems/3396-minimum-number-of-operations-to-make-elements-in-array-distinct/solution.cpp
// Link: https://leetcode.cn/problems/minimum-number-of-operations-to-make-elements-in-array-distinct/
// Level: Easy
// Date: 2025-04-08

class Solution {
    public:
        int minimumOperations(vector<int>& nums) {
            int res = 0;
            int n = nums.size();
            unordered_set<int> s;
            for (int i = n - 1; i >= 0; i--) {
                if (s.find(nums[i]) != s.end()) {
                    res = i / 3 + 1;
                    break;
                }
                s.insert(nums[i]);
            }
            return res;
        }
};

File: problems/3396-minimum-number-of-operations-to-make-elements-in-array-distinct/solution.py
# Link: https://leetcode.cn/problems/minimum-number-of-operations-to-make-elements-in-array-distinct/
# Level: Easy
# Date: 2025-04-08

class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        res = 0
        n = len(nums)
        s = set()
        for i in range(n - 1, -1, -1):
            if nums[i] in s:
                res = i // 3 + 1
                break
            else:
                s.add(nums[i])
        return res